utils.h

```diff
@@ -53,6 +53,7 @@
 #define STATIC_ASSERT(cond) \
 	static uint8_t CONCATENATE(_cmus_unused_, __LINE__)[2*(cond) - 1] UNUSED
 
+#define CLEANUP(f) __attribute__((cleanup(f)))
```

Comment:
Again, `__attribute__` is GCC-specific. Better avoid it, and use standard `goto err` pattern.